File: Java/labs/lab11_Facade/task_2_1 - solution/src/VacationFacade.java
import java.time.LocalDate;

public class VacationFacade {

  public void book(LocalDate startDate, LocalDate endDate) {
    Flight flight = new Flight();
    flight.bookOutwardJourney(startDate);
    flight.bookReturnJourney(endDate);

    Hotel hotel = new Hotel();
    hotel.book(startDate, endDate);

    CarRental carRental = new CarRental();
    carRental.book(startDate, endDate);
  }

}

File: Cpp/lab11_Facade/task_2_1 - solution/src/vacation_facade.cpp
#include "vacation_facade.hpp"


VacationFacade::VacationFacade() {}

void VacationFacade::book(boost::gregorian::date const& startDate, boost::gregorian::date const& endDate)
{
  Flight flight;
  flight.bookOutwardJourney(startDate);
  flight.bookReturnJourney(endDate);

  Hotel hotel;
  hotel.book(startDate, endDate);

  CarRental carRental;
  carRental.book(startDate, endDate);
}
